pandas/tests/frame/test_apply.py

```diff
@@ -718,12 +718,46 @@ def apply_list(row):
 
     def test_apply_noreduction_tzaware_object(self):
         # https://github.com/pandas-dev/pandas/issues/31505
-        df = pd.DataFrame({"foo": [pd.Timestamp("2020", tz="UTC")]}, dtype="object")
+        df = pd.DataFrame(
+            {"foo": [pd.Timestamp("2020", tz="UTC")]}, dtype="datetime64[ns, UTC]"
+        )
         result = df.apply(lambda x: x)
         tm.assert_frame_equal(result, df)
         result = df.apply(lambda x: x.copy())
         tm.assert_frame_equal(result, df)
 
+    def test_apply_function_runs_once(self):
+        # https://github.com/pandas-dev/pandas/issues/30815
+        def non_reducing_func_with_state(row):
```

Comment:
I think would be clearer if you use unittest.mock for the call_count checking. Probably don't need to construct a frame / series from the result

Reply:
Actually was my original idea, but I saw it's not used anywhere. So I thought there might be a reason

Reply:
Ok so as i suspected unittest.mock fails the CI,
`Check that unittest.mock is not used (pytest builtin monkeypatch fixture should be used instead)`

I couldn't find a way to call_count using the builtin monkeypatch, it is available in [this plugin](https://github.com/pytest-dev/pytest-mock).

How should we solve this?